src/include/optimizer/planner.h

```diff
@@ -29,6 +29,13 @@ typedef PlannedStmt *(*planner_hook_type) (Query *parse,
 										   ParamListInfo boundParams);
 extern PGDLLIMPORT planner_hook_type planner_hook;
 
+typedef struct HintState HintState;
+
+/* plan_hint_hook generates HintState by parsing a Query. */
+typedef HintState *(*plan_hint_hook_type) (Query *parse);
+extern PGDLLIMPORT plan_hint_hook_type plan_hint_hook;
```

Comment:
Maybe it is better to move all this hint related stuff into `src/include/optimizer/hints.h`? What do you think?

Reply:
That looks like a good idea, but the question is whether the planner hint hook is more related to the planner or the hint. In planner.h, it's declared near the planner_hook, which looks similar. It's also defined close to planner_hook, although plan_hint_hook is only used by Orca for now.

If we declare it in hints.h, it would be logical to find a different place to put the plan_hint_hook definition. For now, I would leave it here.